src/panel/direction/DirectionPanel.jsx

```diff
@@ -410,8 +409,8 @@ export default class DirectionPanel extends React.Component {
                 </ShareMenu>,
               ]}
               onTransitionEnd={(prevSize, size) => {
-                if (prevSize === 'maximized' && size === 'default' && activeRouteId >= 0) {
-                  fire('set_main_route', { routeId: activeRouteId, fitView: true });
+                if (prevSize === 'maximized' && size === 'default' && this.sanitizeSelected() >= 0) {
```

Comment:
To avoid calling `sanitizeSelected` so many times in the render (and simplify you diff), you could declare `activeRouteId` once at the top.